lib/pos_blockchain.go

```diff
@@ -240,10 +241,19 @@ func (bc *Blockchain) validateAncestorsExist(desoBlock *MsgDeSoBlock) (_missingB
 	return nil, errors.New("IMPLEMENT ME")
 }
 
-// addBlockToBlockIndex adds the block to the block index.
+// addBlockToBlockIndex adds the block to the block index and uncommitted blocks map.
 func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me.
-	return errors.New("IMPLEMENT ME")
+	hash, err := desoBlock.Hash()
+	if err != nil {
+		return errors.Wrapf(err, "addBlockToBlockIndex: Problem hashing block %v", desoBlock)
+	}
+	// Need to get parent block node from block index
+	prevBlock := bc.blockIndex[*desoBlock.Header.PrevBlockHash]
+	// What should the block status be here? Validated?
```

Comment:
I don't have a good answer for you here. Can you take a look through what all of the block status enum values mean? It would be great if you can add comments for them in the enum definition so it's clear to us in the future too.